src/_util.cpp

```diff
@@ -775,6 +783,9 @@ void adjust_font_options(cairo_t* cr)
       : aa.ptr() == Py_False ? CAIRO_ANTIALIAS_NONE
       : aa.cast<cairo_antialias_t>());
   }
+  auto const& variations = *static_cast<std::string*>(
+    cairo_font_face_get_user_data(font_face, &detail::VARIATIONS_KEY));
+  cairo_font_options_set_variations(options, variations.c_str());
```

Comment:
This is absent on cairo<1.16, so it needs to be loaded at runtime.  See e.g. the loading of cairo_tag_begin in _util.h; you can put cairo_font_options_set_variations after it (let's say with the same backport comment but whatever header is appropriate and version 1.16).  At runtime, if the function is not available, emit a warning; again see how it's done for cairo_tag_begin.